dht/token.cc

```diff
@@ -286,4 +286,35 @@ dht::token token::from_int64(int64_t i) {
     return {kind::key, i};
 }
 
+static
+dht::token find_first_token_for_shard_in_not_wrap_around_range(const dht::sharder& sharder, dht::token start, dht::token end, size_t shard_idx) {
+    // Invariant start < end
+    // It is guaranteed that start is not MAX_INT64 because end is greater
+    auto t = dht::token::from_int64(dht::token::to_int64(start) + 1);
+    if (sharder.shard_of(t) != shard_idx) {
+        t = sharder.token_for_next_shard(t, shard_idx);
+    }
+    return std::min(t, end);
+}
+
+dht::token find_first_token_for_shard(
+        const dht::sharder& sharder, dht::token start, dht::token end, size_t shard_idx) {
+    if (start < end) { // Not a wrap around token range
```

Comment:
start == end _is_ a wrap around range, so this is correct. But please add a comment noting it.

Reply:
Yes. Done.